packages/iov-bns-governance/src/index.ts

```diff
@@ -0,0 +1,137 @@
+  REMOVE_VALIDATOR,
+  ACCESS_GUARANTEE_FUND,
+  DISTRIBUTE_FUNDS,
+  AMEND_PROTOCOL,
+}
+
+export interface AddCommitteeMemberOptions {
+  readonly type: ProposalType.ADD_COMMITTEE_MEMBER;
+  readonly committee: CommitteeId;
+  readonly address: Address;
+}
+
+export interface RemoveCommitteeMemberOptions {
+  readonly type: ProposalType.REMOVE_COMMITTEE_MEMBER;
+  readonly committee: CommitteeId;
+  readonly address: Address;
+}
+
+export interface AmendCommitteeThresholdOptions {
+  readonly type: ProposalType.AMEND_COMMITTEE_THRESHOLD;
+  readonly committee: CommitteeId;
+  readonly threshold: Fraction;
+}
+
+export interface AmendCommitteeQuorumOptions {
+  readonly type: ProposalType.AMEND_COMMITTEE_QUORUM;
+  readonly committee: CommitteeId;
+  readonly quorum: Fraction | null;
+}
+
+export interface AddValidatorOptions {
+  readonly type: ProposalType.ADD_VALIDATOR;
+  readonly address: Address;
+}
+
+export interface RemoveValidatorOptions {
+  readonly type: ProposalType.REMOVE_VALIDATOR;
+  readonly address: Address;
+}
+
+export interface AccessGuaranteeFundOptions {
+  readonly type: ProposalType.ACCESS_GUARANTEE_FUND;
+  readonly amount: Amount;
+}
+
+export interface DistributeFundsOptions {
+  readonly type: ProposalType.DISTRIBUTE_FUNDS;
+  readonly recipients: readonly {
+    readonly address: Address;
+    readonly weight: number;
+  }[];
+}
+
+export interface AmendProtocolOptions {
+  readonly type: ProposalType.AMEND_PROTOCOL;
+  readonly text: string;
+}
+
+export type ProposalOptions =
+  | AddCommitteeMemberOptions
+  | RemoveCommitteeMemberOptions
+  | AmendCommitteeThresholdOptions
+  | AmendCommitteeQuorumOptions
+  | AddValidatorOptions
+  | RemoveValidatorOptions
+  | AccessGuaranteeFundOptions
+  | DistributeFundsOptions
+  | AmendProtocolOptions;
+
+export interface GovernorOptions {
+  readonly connection: BnsConnection;
+  readonly profile: UserProfile;
+}
+
+export class Governor {
+  private readonly connection: BnsConnection;
+  private readonly profile: UserProfile;
+
+  public constructor({ connection, profile }: GovernorOptions) {
+    this.connection = connection;
+    this.profile = profile;
+  }
+
+  /* Wraps BnsConnection.getElectorates, filtering for electorates this profile has membership of */
+  public async getElectorates(): Promise<readonly Electorate[]>;
+
+  /* Wraps BnsConnection.getProposals, filtering for proposals this profile can vote on */
+  public async getProposals(): Promise<readonly Proposal[]>;
+
+  /* 1. Constructs a CreateProposalTx using the options and a default fee
+   * 2. Signs the transaction using the profile
+   * 3. Posts the transaction using the connection
+   * 4. Resolves to the PostTxResponse
+   *
+   * Currently we only support creating text resolutions, so this would just format the relevant information
+   * into a standardised string for each proposal type.
+   */
+  public async submitProposal(options: ProposalOptions): Promise<PostTxResponse>;
```

Comment:
This needs to be split, since
1. is done by the voting app
2. is done by the browser extension
3. is done by the browser extension
4. only the transaction ID is sent back to the voting app

So I think this could be a transaction creator instead